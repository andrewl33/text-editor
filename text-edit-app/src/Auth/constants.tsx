// dashboard
// add file to account
// remove file from account
// add collection to account
// remove collection from account

// auth
// create account
// log in
// log out
export const CREATE_ACCOUNT_REQUEST = "CREATE_ACCOUNT_REQUEST";
export type CREATE_ACCOUNT_REQUEST = typeof CREATE_ACCOUNT_REQUEST;

export const CREATE_ACCOUNT_SUCCESS = "CREATE_ACCOUNT_SUCCESS";
export type CREATE_ACCOUNT_SUCCESS = typeof CREATE_ACCOUNT_SUCCESS;

export const CREATE_ACCOUNT_FAILURE = "CREATE_ACCOUNT_FAILURE";
export type CREATE_ACCOUNT_FAILURE = typeof CREATE_ACCOUNT_FAILURE;

export const LOG_IN_REQUEST = "LOG_IN_REQUEST";
export type LOG_IN_REQUEST = typeof LOG_IN_REQUEST;

export const LOG_IN_SUCCESS = "LOG_IN_SUCCESS";
export type LOG_IN_SUCCESS = typeof LOG_IN_SUCCESS;

export const LOG_IN_FAILURE = "LOG_IN_FAILURE";
export type LOG_IN_FAILURE = typeof LOG_IN_FAILURE;

export const LOG_OUT_REQUEST = "LOG_OUT_REQUEST";
export type LOG_OUT_REQUEST = typeof LOG_OUT_REQUEST;

export const LOG_OUT_SUCCESS = "LOG_OUT_SUCCESS";
export type LOG_OUT_SUCCESS = typeof LOG_OUT_SUCCESS;

export const LOG_OUT_FAILURE = "LOG_OUT_FAILURE";
export type LOG_OUT_FAILURE = typeof LOG_OUT_FAILURE;

export const LOG_IN_PROMPT = "LOG_IN_PROMPT";
export type LOG_IN_PROMPT = typeof LOG_IN_PROMPT;

export const CLOSE_PROMPT = "CLOSE_PROMPT";
export type CLOSE_PROMPT = typeof CLOSE_PROMPT;

export const UPDATE_TOKEN = "UPDATE_TOKEN";
export type UPDATE_TOKEN = typeof UPDATE_TOKEN;

export const GET_DASHBOARD_REQUEST = "GET_DASHBOARD_REQUEST";
export type GET_DASHBOARD_REQUEST = typeof GET_DASHBOARD_REQUEST;

export const GET_DASHBOARD_SUCCESS = "GET_DASHBOARD_SUCCESS";
export type GET_DASHBOARD_SUCCESS = typeof GET_DASHBOARD_SUCCESS;

export const GET_DASHBOARD_FAILURE = "GET_DASHBOARD_FAILURE";
export type GET_DASHBOARD_FAILURE = typeof GET_DASHBOARD_FAILURE;
