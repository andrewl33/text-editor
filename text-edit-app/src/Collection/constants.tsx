export const NEW_COLLECTION_REQUEST = "NEW_COLLECTION_REQUEST";
export type NEW_COLLECTION_REQUEST = typeof NEW_COLLECTION_REQUEST;

export const NEW_COLLECTION_SUCCESS = "NEW_COLLECTION_SUCCESS";
export type NEW_COLLECTION_SUCCESS = typeof NEW_COLLECTION_SUCCESS;

export const NEW_COLLECTION_FAILURE = "NEW_COLLECTION_FAILURE";
export type NEW_COLLECTION_FAILURE = typeof NEW_COLLECTION_FAILURE;

export const GET_COLLECTION_REQUEST = "GET_COLLECTION_REQUEST";
export type GET_COLLECTION_REQUEST = typeof GET_COLLECTION_REQUEST;

export const GET_COLLECTION_SUCCESS = "GET_COLLECTION_SUCCESS";
export type GET_COLLECTION_SUCCESS = typeof GET_COLLECTION_SUCCESS;

export const GET_COLLECTION_FAILURE = "GET_COLLECTION_FAILURE";
export type GET_COLLECTION_FAILURE = typeof GET_COLLECTION_FAILURE;

export const GET_COLLECTION_AUTH = "GET_COLLECTION_AUTH";
export type GET_COLLECTION_AUTH = typeof GET_COLLECTION_AUTH;

export const AUTH_COLLECTION_REQUEST = "AUTH_COLLECTION_REQUEST";
export type AUTH_COLLECTION_REQUEST = typeof AUTH_COLLECTION_REQUEST;

export const AUTH_COLLECTION_SUCCESS = "AUTH_COLLECTION_SUCCESS";
export type AUTH_COLLECTION_SUCCESS = typeof AUTH_COLLECTION_SUCCESS;

export const AUTH_COLLECTION_FAILURE = "AUTH_COLLECTION_FAILURE";
export type AUTH_COLLECTION_FAILURE = typeof AUTH_COLLECTION_FAILURE;

export const LOCK_COLLECTION_REQUEST = "LOCK_COLLECTION_REQUEST";
export type LOCK_COLLECTION_REQUEST = typeof LOCK_COLLECTION_REQUEST;

export const LOCK_COLLECTION_SUCCESS = "LOCK_COLLECTION_SUCCESS";
export type LOCK_COLLECTION_SUCCESS = typeof LOCK_COLLECTION_SUCCESS;

export const LOCK_COLLECTION_FAILURE = "LOCK_COLLECTION_FAILURE";
export type LOCK_COLLECTION_FAILURE = typeof LOCK_COLLECTION_FAILURE;

export const ADD_FILE_TO_COLLECTION_REQUEST = "ADD_FILE_TO_COLLECTION_REQUEST";
export type ADD_FILE_TO_COLLECTION_REQUEST = typeof ADD_FILE_TO_COLLECTION_REQUEST;

export const ADD_FILE_TO_COLLECTION_SUCCESS = "ADD_FILE_TO_COLLECTION_SUCCESS";
export type ADD_FILE_TO_COLLECTION_SUCCESS = typeof ADD_FILE_TO_COLLECTION_SUCCESS;

export const ADD_FILE_TO_COLLECTION_FAILURE = "ADD_FILE_TO_COLLECTION_FAILURE";
export type ADD_FILE_TO_COLLECTION_FAILURE = typeof ADD_FILE_TO_COLLECTION_FAILURE;

export const REMOVE_FILE_FROM_COLLECTION_REQUEST =
  "REMOVE_FILE_FROM_COLLECTION_REQUEST";
export type REMOVE_FILE_FROM_COLLECTION_REQUEST = typeof REMOVE_FILE_FROM_COLLECTION_REQUEST;

export const REMOVE_FILE_FROM_COLLECTION_SUCCESS =
  "REMOVE_FILE_FROM_COLLECTION_SUCCESS";
export type REMOVE_FILE_FROM_COLLECTION_SUCCESS = typeof REMOVE_FILE_FROM_COLLECTION_SUCCESS;

export const REMOVE_FILE_FROM_COLLECTION_FAILURE =
  "REMOVE_FILE_FROM_COLLECTION_FAILURE";
export type REMOVE_FILE_FROM_COLLECTION_FAILURE = typeof REMOVE_FILE_FROM_COLLECTION_FAILURE;

export const CHANGE_COLLECTION_NAME_REQUEST = "CHANGE_COLLECTION_NAME_REQUEST";
export type CHANGE_COLLECTION_NAME_REQUEST = typeof CHANGE_COLLECTION_NAME_REQUEST;

export const CHANGE_COLLECTION_NAME_SUCCESS = "CHANGE_COLLECTION_NAME_SUCCESS";
export type CHANGE_COLLECTION_NAME_SUCCESS = typeof CHANGE_COLLECTION_NAME_SUCCESS;

export const CHANGE_COLLECTION_NAME_FAILURE = "CHANGE_COLLECTION_NAME_FAILURE";
export type CHANGE_COLLECTION_NAME_FAILURE = typeof CHANGE_COLLECTION_NAME_FAILURE;

export const ADD_USER_TO_COLLECTION_REQUEST = "ADD_USER_TO_COLLECTION_REQUEST";
export type ADD_USER_TO_COLLECTION_REQUEST = typeof ADD_USER_TO_COLLECTION_REQUEST;

export const ADD_USER_TO_COLLECTION_SUCCESS = "ADD_USER_TO_COLLECTION_SUCCESS";
export type ADD_USER_TO_COLLECTION_SUCCESS = typeof ADD_USER_TO_COLLECTION_SUCCESS;

export const ADD_USER_TO_COLLECTION_FAILURE = "ADD_USER_TO_COLLECTION_FAILURE";
export type ADD_USER_TO_COLLECTION_FAILURE = typeof ADD_USER_TO_COLLECTION_FAILURE;

export const REMOVE_USER_FROM_COLLECTION_REQUEST =
  "REMOVE_USER_FROM_COLLECTION_REQUEST";
export type REMOVE_USER_FROM_COLLECTION_REQUEST = typeof REMOVE_USER_FROM_COLLECTION_REQUEST;

export const REMOVE_USER_FROM_COLLECTION_SUCCESS =
  "REMOVE_USER_FROM_COLLECTION_SUCCESS";
export type REMOVE_USER_FROM_COLLECTION_SUCCESS = typeof REMOVE_USER_FROM_COLLECTION_SUCCESS;

export const REMOVE_USER_FROM_COLLECTION_FAILURE =
  "REMOVE_USER_FROM_COLLECTION_FAILURE";
export type REMOVE_USER_FROM_COLLECTION_FAILURE = typeof REMOVE_USER_FROM_COLLECTION_FAILURE;
