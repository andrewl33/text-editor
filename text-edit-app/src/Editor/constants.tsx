// editor
export const LOCAL_UPDATE_CODE = "LOCAL_UPDATE_CODE";
export type LOCAL_UPDATE_CODE = typeof LOCAL_UPDATE_CODE;

export const UPDATE_CODE_REQUEST = "UPDATE_CODE_REQUEST";
export type UPDATE_CODE_REQUEST = typeof UPDATE_CODE_REQUEST;

export const UPDATE_CODE_SUCCESS = "UPDATE_CODE_SUCCESS";
export type UPDATE_CODE_SUCCESS = typeof UPDATE_CODE_SUCCESS;

export const UPDATE_CODE_FAILURE = "UPDATE_CODE_FAILURE";
export type UPDATE_CODE_FAILURE = typeof UPDATE_CODE_FAILURE;

export const CHANGED_CODE = "CHANGED_CODE";
export type CHANGED_CODE = typeof CHANGED_CODE;

export const NEW_TEXT_REQUEST = "NEW_TEXT_REQUEST";
export type NEW_TEXT_REQUEST = typeof NEW_TEXT_REQUEST;

export const NEW_TEXT_SUCCESS = "NEW_TEXT_SUCCESS";
export type NEW_TEXT_SUCCESS = typeof NEW_TEXT_SUCCESS;

export const NEW_TEXT_FAILURE = "NEW_TEXT_FAILURE";
export type NEW_TEXT_FAILURE = typeof NEW_TEXT_FAILURE;

export const GET_TEXT_REQUEST = "GET_TEXT_REQUEST";
export type GET_TEXT_REQUEST = typeof GET_TEXT_REQUEST;

export const GET_TEXT_SUCCESS = "GET_TEXT_SUCCESS";
export type GET_TEXT_SUCCESS = typeof GET_TEXT_SUCCESS;

export const GET_TEXT_FAILURE = "GET_TEXT_FAILURE";
export type GET_TEXT_FAILURE = typeof GET_TEXT_FAILURE;

export const GET_TEXT_AUTH = "GET_TEXT_AUTH";
export type GET_TEXT_AUTH = typeof GET_TEXT_AUTH;

export const AUTH_FILE_REQUEST = "AUTH_FILE_REQUEST";
export type AUTH_FILE_REQUEST = typeof AUTH_FILE_REQUEST;

export const AUTH_FILE_SUCCESS = "AUTH_FILE_SUCCESS";
export type AUTH_FILE_SUCCESS = typeof AUTH_FILE_SUCCESS;

export const AUTH_FILE_FAILURE = "AUTH_FILE_FAILURE";
export type AUTH_FILE_FAILURE = typeof AUTH_FILE_FAILURE;

export const ADD_TAG_REQUEST = "ADD_TAG_REQUEST";
export type ADD_TAG_REQUEST = typeof ADD_TAG_REQUEST;

export const ADD_TAG_SUCCESS = "ADD_TAG_SUCCESS";
export type ADD_TAG_SUCCESS = typeof ADD_TAG_SUCCESS;

export const ADD_TAG_FAILURE = "ADD_TAG_FAILURE";
export type ADD_TAG_FAILURE = typeof ADD_TAG_FAILURE;

export const REMOVE_TAG_REQUEST = "REMOVE_TAG_REQUEST";
export type REMOVE_TAG_REQUEST = typeof REMOVE_TAG_REQUEST;

export const REMOVE_TAG_SUCCESS = "REMOVE_TAG_SUCCESS";
export type REMOVE_TAG_SUCCESS = typeof REMOVE_TAG_SUCCESS;

export const REMOVE_TAG_FAILURE = "REMOVE_TAG_FAILURE";
export type REMOVE_TAG_FAILURE = typeof REMOVE_TAG_FAILURE;

export const CHANGE_FILE_NAME_REQUEST = "CHANGE_FILE_NAME_REQUEST";
export type CHANGE_FILE_NAME_REQUEST = typeof CHANGE_FILE_NAME_REQUEST;

export const CHANGE_FILE_NAME_SUCCESS = "CHANGE_FILE_NAME_SUCCESS";
export type CHANGE_FILE_NAME_SUCCESS = typeof CHANGE_FILE_NAME_SUCCESS;

export const CHANGE_FILE_NAME_FAILURE = "CHANGE_FILE_NAME_FAILURE";
export type CHANGE_FILE_NAME_FAILURE = typeof CHANGE_FILE_NAME_FAILURE;

export const LOCK_TEXT_REQUEST = "LOCK_TEXT_REQUEST";
export type LOCK_TEXT_REQUEST = typeof LOCK_TEXT_REQUEST;

export const ADD_USER_TO_FILE_REQUEST = "ADD_USER_TO_FILE_REQUEST";
export type ADD_USER_TO_FILE_REQUEST = typeof ADD_USER_TO_FILE_REQUEST;

export const ADD_USER_TO_FILE_SUCCESS = "ADD_USER_TO_FILE_SUCCESS";
export type ADD_USER_TO_FILE_SUCCESS = typeof ADD_USER_TO_FILE_SUCCESS;

export const ADD_USER_TO_FILE_FAILURE = "ADD_USER_TO_FILE_FAILURE";
export type ADD_USER_TO_FILE_FAILURE = typeof ADD_USER_TO_FILE_FAILURE;

export const REMOVE_USER_FROM_FILE_REQUEST = "REMOVE_USER_FROM_FILE_REQUEST";
export type REMOVE_USER_FROM_FILE_REQUEST = typeof REMOVE_USER_FROM_FILE_REQUEST;

export const REMOVE_USER_FROM_FILE_SUCCESS = "REMOVE_USER_FROM_FILE_SUCCESS";
export type REMOVE_USER_FROM_FILE_SUCCESS = typeof REMOVE_USER_FROM_FILE_SUCCESS;

export const REMOVE_USER_FROM_FILE_FAILURE = "REMOVE_USER_FROM_FILE_FAILURE";
export type REMOVE_USER_FROM_FILE_FAILURE = typeof REMOVE_USER_FROM_FILE_FAILURE;

export const LOCK_TEXT_SUCCESS = "LOCK_TEXT_SUCCESS";
export type LOCK_TEXT_SUCCESS = typeof LOCK_TEXT_SUCCESS;

export const LOCK_TEXT_FAILURE = "LOCK_TEXT_FAILURE";
export type LOCK_TEXT_FAILURE = typeof LOCK_TEXT_FAILURE;
