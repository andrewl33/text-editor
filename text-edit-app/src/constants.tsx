export const NEW_TEXT = 'NEW_TEXT';
export type NEW_TEXT = typeof NEW_TEXT;

export const LOCK_TEXT = 'LOCK_TEXT';
export type LOCK_TEXT = typeof LOCK_TEXT;

export const SHARE_LINK = 'SHARE_LINK';
export type SHARE_LINK = typeof SHARE_LINK;