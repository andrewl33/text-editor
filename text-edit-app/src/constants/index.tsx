// header
export const NEW_TEXT = 'NEW_TEXT';
export type NEW_TEXT = typeof NEW_TEXT;

export const LOCK_TEXT = 'LOCK_TEXT';
export type LOCK_TEXT = typeof LOCK_TEXT;

export const SHARE_LINK = 'SHARE_LINK';
export type SHARE_LINK = typeof SHARE_LINK;

// editor
export const UPDATE_CODE = 'UPDATE_CODE';
export type UPDATE_CODE = typeof UPDATE_CODE;

export const CHANGED_CODE = 'CHANGED_CODE';
export type CHANGED_CODE = typeof CHANGED_CODE;