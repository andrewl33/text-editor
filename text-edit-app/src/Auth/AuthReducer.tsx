import { AuthAction } from "./AuthAction";

import { CLOSE_ALERT } from "../constants";
import { AuthStoreState } from "../types";
import {
  CLOSE_PROMPT,
  CREATE_ACCOUNT_FAILURE,
  CREATE_ACCOUNT_REQUEST,
  CREATE_ACCOUNT_SUCCESS,
  LOG_IN_FAILURE,
  LOG_IN_PROMPT,
  LOG_IN_REQUEST,
  LOG_IN_SUCCESS,
  LOG_OUT_FAILURE,
  LOG_OUT_REQUEST,
  LOG_OUT_SUCCESS
} from "./constants";

export const initialState: AuthStoreState = {
  loggedIn: false,
  accountName: undefined,
  dashboard: undefined,
  openAlert: false,
  alertMessage: undefined,
  authPrompt: false
};

export const authReducer = (
  state: AuthStoreState = initialState,
  action: AuthAction
) => {
  switch (action.type) {
    case CREATE_ACCOUNT_REQUEST:
      return { ...state };
    case CREATE_ACCOUNT_SUCCESS:
      return {
        ...state,
        loggedIn: true,
        accountName: action.payload && action.payload.accountName
      };
    case CREATE_ACCOUNT_FAILURE:
      return { ...state };

    case LOG_IN_REQUEST:
      return { ...state };
    case LOG_IN_SUCCESS:
      return {
        ...state,
        loggedIn: true,
        accountName: action.payload && action.payload.accountName,
        dashboard: action.payload && action.payload.dashboard,
        authPrompt: false
      };
    case LOG_IN_FAILURE:
      return { ...state };

    case LOG_OUT_REQUEST:
      return { ...state };
    case LOG_OUT_SUCCESS:
      return { ...initialState };
    case LOG_OUT_FAILURE:
      return { ...state };

    case LOG_IN_PROMPT:
      return { ...state, authPrompt: true };

    case CLOSE_PROMPT:
      return { ...state, authPrompt: false };

    case CLOSE_ALERT:
      return { ...state, openAlert: false };

    default:
      return state;
  }
};
