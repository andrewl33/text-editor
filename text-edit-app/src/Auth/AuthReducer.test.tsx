import * as constants from "../constants";
import { authReducer } from "./AuthReducer";

describe("AuthReducer", () => {
  test("it should return initial state", () => {
    expect(authReducer(undefined, {} as any)).toMatchSnapshot();
  });

  test("it should handle UPDATE_TOKEN", () => {
    expect(
      authReducer(undefined, {
        type: constants.UPDATE_TOKEN,
        payload: { success: true, token: "Bearer FakeTokenToken" }
      })
    ).toMatchSnapshot();
  });

  describe("createAccount", () => {
    test("it should handle CREATE_ACCOUNT_REQUEST", () => {
      expect(
        authReducer(undefined, { type: constants.CREATE_ACCOUNT_REQUEST })
      ).toMatchSnapshot();
    });

    test("it should handle CREATE_ACCOUNT_SUCCESS", () => {
      expect(
        authReducer(undefined, {
          type: constants.CREATE_ACCOUNT_SUCCESS,
          payload: { accountName: "testName", success: true }
        })
      ).toMatchSnapshot();
    });

    test("it should handle CREATE_ACCOUNT_FAILURE", () => {
      expect(
        authReducer(undefined, { type: constants.CREATE_ACCOUNT_FAILURE })
      ).toMatchSnapshot();
    });
  });

  describe("logIn", () => {
    test("it should handle LOG_IN_REQUEST", () => {
      expect(
        authReducer(undefined, { type: constants.LOG_IN_REQUEST })
      ).toMatchSnapshot();
    });

    test("it should handle LOG_IN_SUCCESS", () => {
      expect(
        authReducer(undefined, {
          type: constants.LOG_IN_SUCCESS,
          payload: { accountName: "testName", success: true }
        })
      ).toMatchSnapshot();
    });

    test("it should handle LOG_IN_FAILURE", () => {
      expect(
        authReducer(undefined, { type: constants.LOG_IN_FAILURE })
      ).toMatchSnapshot();
    });
  });

  describe("getDashboard", () => {
    test("it should handle GET_DASHBOARD_REQUEST", () => {
      expect(
        authReducer(undefined, { type: constants.GET_DASHBOARD_REQUEST })
      ).toMatchSnapshot();
    });

    test("it should handle GET_DASHBOARD_SUCCESS", () => {
      expect(
        authReducer(undefined, {
          type: constants.GET_DASHBOARD_SUCCESS,
          payload: { dashboard: { collections: [], files: [] }, success: true }
        })
      ).toMatchSnapshot();
    });

    test("it should handle GET_DASHBOARD_FAILURE", () => {
      expect(
        authReducer(undefined, { type: constants.GET_DASHBOARD_FAILURE })
      ).toMatchSnapshot();
    });
  });

  describe("logOut", () => {
    test("it should handle LOG_OUT_REQUEST", () => {
      expect(
        authReducer(undefined, { type: constants.LOG_OUT_REQUEST })
      ).toMatchSnapshot();
    });

    test("it should handle LOG_OUT_SUCCESS", () => {
      expect(
        authReducer(undefined, {
          type: constants.LOG_OUT_SUCCESS
        })
      ).toMatchSnapshot();
    });

    test("it should handle LOG_OUT_FAILURE", () => {
      expect(
        authReducer(undefined, { type: constants.LOG_OUT_FAILURE })
      ).toMatchSnapshot();
    });
  });

  describe("authPrompt", () => {
    test("it should handle LOG_IN_PROMPT", () => {
      expect(
        authReducer(undefined, { type: constants.LOG_IN_PROMPT })
      ).toMatchSnapshot();
    });

    test("it should handle CLOSE_PROMPT", () => {
      expect(
        authReducer(undefined, {
          type: constants.CLOSE_PROMPT
        })
      ).toMatchSnapshot();
    });
  });
});
