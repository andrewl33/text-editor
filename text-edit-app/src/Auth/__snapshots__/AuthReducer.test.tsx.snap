// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AuthReducer authPrompt it should handle CLOSE_PROMPT 1`] = `
Object {
  "accountName": undefined,
  "alertMessage": undefined,
  "authPrompt": false,
  "dashboard": undefined,
  "loggedIn": false,
  "openAlert": false,
  "token": undefined,
}
`;

exports[`AuthReducer authPrompt it should handle LOG_IN_PROMPT 1`] = `
Object {
  "accountName": undefined,
  "alertMessage": undefined,
  "authPrompt": true,
  "dashboard": undefined,
  "loggedIn": false,
  "openAlert": false,
  "token": undefined,
}
`;

exports[`AuthReducer createAccount it should handle CREATE_ACCOUNT_FAILURE 1`] = `
Object {
  "accountName": undefined,
  "alertMessage": undefined,
  "authPrompt": false,
  "dashboard": undefined,
  "loggedIn": false,
  "openAlert": false,
  "token": undefined,
}
`;

exports[`AuthReducer createAccount it should handle CREATE_ACCOUNT_REQUEST 1`] = `
Object {
  "accountName": undefined,
  "alertMessage": undefined,
  "authPrompt": false,
  "dashboard": undefined,
  "loggedIn": false,
  "openAlert": false,
  "token": undefined,
}
`;

exports[`AuthReducer createAccount it should handle CREATE_ACCOUNT_SUCCESS 1`] = `
Object {
  "accountName": "testName",
  "alertMessage": undefined,
  "authPrompt": false,
  "dashboard": undefined,
  "loggedIn": true,
  "openAlert": false,
  "token": undefined,
}
`;

exports[`AuthReducer getDashboard it should handle GET_DASHBOARD_FAILURE 1`] = `
Object {
  "accountName": undefined,
  "alertMessage": "Not logged in",
  "authPrompt": false,
  "dashboard": undefined,
  "loggedIn": false,
  "openAlert": true,
  "token": undefined,
}
`;

exports[`AuthReducer getDashboard it should handle GET_DASHBOARD_REQUEST 1`] = `
Object {
  "accountName": undefined,
  "alertMessage": undefined,
  "authPrompt": false,
  "dashboard": undefined,
  "loggedIn": false,
  "openAlert": false,
  "token": undefined,
}
`;

exports[`AuthReducer getDashboard it should handle GET_DASHBOARD_SUCCESS 1`] = `
Object {
  "accountName": undefined,
  "alertMessage": undefined,
  "authPrompt": false,
  "dashboard": Object {
    "collections": Array [],
    "files": Array [],
  },
  "loggedIn": false,
  "openAlert": false,
  "token": undefined,
}
`;

exports[`AuthReducer it should handle UPDATE_TOKEN 1`] = `
Object {
  "accountName": undefined,
  "alertMessage": undefined,
  "authPrompt": false,
  "dashboard": undefined,
  "loggedIn": false,
  "openAlert": false,
  "token": "Bearer FakeTokenToken",
}
`;

exports[`AuthReducer it should return initial state 1`] = `
Object {
  "accountName": undefined,
  "alertMessage": undefined,
  "authPrompt": false,
  "dashboard": undefined,
  "loggedIn": false,
  "openAlert": false,
  "token": undefined,
}
`;

exports[`AuthReducer logIn it should handle LOG_IN_FAILURE 1`] = `
Object {
  "accountName": undefined,
  "alertMessage": "Wrong account info",
  "authPrompt": false,
  "dashboard": undefined,
  "loggedIn": false,
  "openAlert": true,
  "token": undefined,
}
`;

exports[`AuthReducer logIn it should handle LOG_IN_REQUEST 1`] = `
Object {
  "accountName": undefined,
  "alertMessage": undefined,
  "authPrompt": false,
  "dashboard": undefined,
  "loggedIn": false,
  "openAlert": false,
  "token": undefined,
}
`;

exports[`AuthReducer logIn it should handle LOG_IN_SUCCESS 1`] = `
Object {
  "accountName": "testName",
  "alertMessage": undefined,
  "authPrompt": false,
  "dashboard": undefined,
  "loggedIn": true,
  "openAlert": false,
  "token": undefined,
}
`;

exports[`AuthReducer logOut it should handle LOG_OUT_FAILURE 1`] = `
Object {
  "accountName": undefined,
  "alertMessage": undefined,
  "authPrompt": false,
  "dashboard": undefined,
  "loggedIn": false,
  "openAlert": false,
  "token": undefined,
}
`;

exports[`AuthReducer logOut it should handle LOG_OUT_REQUEST 1`] = `
Object {
  "accountName": undefined,
  "alertMessage": undefined,
  "authPrompt": false,
  "dashboard": undefined,
  "loggedIn": false,
  "openAlert": false,
  "token": undefined,
}
`;

exports[`AuthReducer logOut it should handle LOG_OUT_SUCCESS 1`] = `
Object {
  "accountName": undefined,
  "alertMessage": undefined,
  "authPrompt": false,
  "dashboard": undefined,
  "loggedIn": false,
  "openAlert": false,
  "token": undefined,
}
`;
